py/server/deephaven/parquet.py

```diff
@@ -125,12 +184,17 @@ class ParquetFileLayout(Enum):
 
 
 def read(
-    path: str,
-    col_instructions: Optional[List[ColumnInstruction]] = None,
-    is_legacy_parquet: bool = False,
-    is_refreshing: bool = False,
-    file_layout: Optional[ParquetFileLayout] = None,
-    table_definition: Union[Dict[str, DType], List[Column], None] = None,
+        path: str,
+        col_instructions: Optional[List[ColumnInstruction]] = None,
+        is_legacy_parquet: bool = False,
+        is_refreshing: bool = False,
+        file_layout: Optional[ParquetFileLayout] = None,
+        table_definition: Union[Dict[str, DType], List[Column], None] = None,
+        aws_region_name: str = None,
+        max_concurrent_requests: int = None,
+        read_ahead_count: int = None,
+        fragment_size: int = None,
+        max_cache_size: int = None,
```

Comment:
It's not obvious to me if we should have these options like this, or use something like `special_instructions: Optional[Any] = None`. I somewhat lean towards the latter. No need to change right now until python review.

```python
from deephaven import parquet
from deephaven import s3

my_table = parquet.read(
    "s3://my-bucket/my-path/file.parquet",
    special_instructions=s3.instructions("us-west-2"),
)

```